source/distributed/shared_tria.cc

```diff
@@ -134,6 +134,16 @@ namespace parallel
       this->update_number_cache ();
     }
 
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::
+    copy_triangulation (const dealii::Triangulation<dim, spacedim> &old_tria)
+    {
+      dealii::Triangulation<dim,spacedim>::copy_triangulation (old_tria);
```

Comment:
```
    if (const dealii::parallel::Triangulation<dim,spacedim> *
        old_tria_x = dynamic_cast<const dealii::parallel::Triangulation<dim,spacedim> *>(&old_tria))
      {
        mpi_communicator = Utilities::MPI::duplicate_communicator (old_tria_x->get_communicator ());
      }
```
We assign a new value to `mpi_communicator`, but we do not free the old value before.